pages/merchant/_reducer.tsx

```diff
@@ -8,41 +8,40 @@ export const ACTIONS = {
   CLEAR_INPUT: "clear-input",
   CREATE_INVOICE: "create-invoice",
   CREATE_NEW_INVOICE: "create-new-invoice",
-  GET_PAYMENT_STATUS: "get-payment-status",
   UPDATE_USERNAME: "update-username",
   UPDATE_WALLET_CURRENCY: "update-wallet-currency",
   SET_AMOUNT_FROM_PARAMS: "set-amount-from-search",
+  PINNED_TO_HOMESCREEN: "pin-to-home-screen",
   BACK: "back-by-one-history",
 }
 
 export type ACTION_TYPE = {
   type: string
-  payload?: string | string[] | (() => void) | undefined
+  payload?: string | string[] | (() => void) | boolean | undefined
 }
 
 function reducer(state: React.ComponentState, { type, payload }: ACTION_TYPE) {
   switch (type) {
     case ACTIONS.ADD_DIGIT:
-      if (payload == "0" && state.currentAmount == "0") return { state }
-      if (payload === "." && state.currentAmount.includes(".")) return { state }
-      if (state.currentAmount == undefined) return { state }
-      if (state.currentAmount?.length >= MAX_INPUT_VALUE_LENGTH) return { state }
+      if (payload == "0" && state.currentAmount == "0") return { ...state }
```

Comment:
If you dont want the page to rerender you should ```return state``` not ```return {...state}``` which creates a new object.

Reply:
This doesn't trigger a re-render

Reply:
Its returning a new state object. Doesn't that force anything that is relying on that state to rerender?

Reply:
For example if another hook had the reducer state in its dependency array, even if the state had the same values, the reference to the state object would be different causing the hook to rerun even though its dependency did not actually change.

Reply:
You are right as regards the referencing of the state object. The goal is to always have a copy of the old state when there is a mismatch or validation error. So we persist the old state in the reducer.

Reply:
the re-render is only triggered when there's validation error or mismatch